tokio/src/runtime/handle.rs

```diff
@@ -201,6 +201,18 @@ impl Handle {
         let _ = self.blocking_spawner.spawn(task, &self);
         handle
     }
+
+    /// TODO: write docs if this is a good direction
+    pub fn block_on<F: Future>(&self, future: F) -> F::Output {
+        // Enter the **runtime** context. This configures spawning, the current I/O driver, ...
+        let _rt_enter = self.enter();
```

Comment:
can enter check if the runtime is still alive?

Reply:
I think you opened a different issue, but that is someone of an orthogonal concern :)